core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -294,12 +294,14 @@ private int sendOrProcessProvisionedRequests() {
                 dispatcherRegistry.send(Void.class, dataRequest, process::getId);
             } else {
                 var response = dataFlowManager.initiate(dataRequest);
-                if (ResponseStatus.ERROR_RETRY == response.getFailure().status()) {
+                if (response.getFailure() != null &&
```

Comment:
the `failed()` method is more easy to read.

Reply:
If this is ok for you I'd leave it like this. I just added a null check here and cannot tell, why the status is compared to ERROR_RETRY instead of calling the failed() method.

Reply:
Probably it was my fault, the way to test if a result is failed is to call the `failed` method. Since the result is failed, the failure is not null, so I would prefer that over the null checking

Reply:
yes

Reply:
@ndr-brt done :)